documentation/api/query/v4/upgrading-from-v3.markdown

```diff
@@ -0,0 +1,128 @@
+---
+title: "PuppetDB 3.0 » API » v4 » Upgrading to PuppetDB 3.0"
+layout: default
+canonical: "/puppetdb/latest/api/query/v4/upgrading-from-v3.html"
+---
+
+This document describes changes that users will need to be aware of
+to make their code compliant with the breaking changes in PuppetDB 3.0. Most of these
+changes are around the v4 query API, which has been marked 'experimental' since
+2.0.0 but will be the only API available in 3.0. This is not a changelog
+for PuppetDB 3.0, since some of the changes described are present in 2.x.x.
```

Comment:
'Please note this document focuses on API changes only, if you're looking for a proper changelog, see the release notes here' or something.